libs/api/ebpf_api.cpp

```diff
@@ -149,11 +152,12 @@ _get_wstring_from_string(std::string& text) noexcept(false)
     return wide;
 }
 
-inline static ebpf_map_t*
-_get_ebpf_map_from_handle(ebpf_handle_t map_handle) noexcept
+_Requires_lock_not_held_(_ebpf_state_mutex) inline static ebpf_map_t* _get_ebpf_map_from_handle(
+    ebpf_handle_t map_handle) noexcept
 {
     EBPF_LOG_ENTRY();
 
+    std::unique_lock lock(_ebpf_state_mutex);
     ebpf_assert(map_handle != ebpf_handle_invalid);
 
     ebpf_map_t* map = nullptr;
```

Comment:
nit: minor suggestion:
```suggestion
    EBPF_LOG_ENTRY();
    ebpf_assert(map_handle != ebpf_handle_invalid);

    std::unique_lock lock(_ebpf_state_mutex);
    ebpf_map_t* map = nullptr;
```